include/bolgenos-ng/memory.hpp

```diff
@@ -2,64 +2,70 @@
 
 #include <bolgenos-ng/int_types.h>
 
+#include <bolgenos-ng/memory_region.hpp>
+
+/// \brief Memory namespace
+///
+/// Namespace contains functionality that is related to high-level memory
+/// management.
 namespace memory {
-/**
-* \brief Aling value down.
-*
-* Align given value down to specified boundary.
-*
-* \param value Value to be aligned.
-* \param boundary Alignment boundary.
-* \return Aligned value.
-*/
-static inline size_t align_down(size_t value, size_t boundary) {
-	return value & ~(boundary - 1);
+
+
+/// \brief Aling specified value down.
+///
+/// Align given value down to specified boundary.
+///
+/// \tparam boundary Alignment boundary.
+/// \tparam ValueType Type of value that is to be aligned.
+/// \param value Value to be aligned.
+/// \return Aligned value.
```

Comment:
:exclamation: Undo changing of doxygen comments style. If it's needed it should be done as separate feature